uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java

```diff
@@ -292,16 +327,111 @@ private final String basicButtonHandler(@NonNull final String resourceButtonId,
             UiAutomatorUtils.handleButtonClick("android:id/button1");
         }
     }
-
-    @Override
-    public void setFlights(@Nullable final String flightsJson) {
-        Logger.i(TAG, "Set Flights..");
+    /**
+     * Gets all the accounts added
+     */
+    public List<String> getAllAccounts(boolean expectingMultipleAccounts) {
         launch();
-        // Make the set flights UI visible on screen
-        UiAutomatorUtils.obtainChildInScrollable("Set Flights");
-        // input flights string in flights input box
-        UiAutomatorUtils.handleInput("com.microsoft.identity.testuserapp:id/editTextFlights", flightsJson);
-        // Click Set Flights button
-        UiAutomatorUtils.handleButtonClick("com.microsoft.identity.testuserapp:id/setFlightsButton");
+
+        final String resourceButtonId = "com.microsoft.identity.testuserapp:id/buttonGetAccounts";
+        final String textId = "AccountName";
+        if (shouldHandleFirstRun) {
+            handleFirstRun(); // handle first run experience
+        }
+        UiAutomatorUtils.handleButtonClick(resourceButtonId);
+        // Look for the dialog box
+        final UiObject dialogBox = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                "android:id/message"
+        );
+        Assert.assertTrue(dialogBox.exists());
+
+        // As of now we are testing only for 2 accounts.
+        List<String> accounts = new ArrayList<>();
+        final String accountName = getDialogBoxText(dialogBox, textId);
+        if (accountName != null) {
+            accounts.add(accountName);
+        }
+        if (expectingMultipleAccounts) {
+            final String accountName2 = getDialogBoxText(dialogBox, textId);
+            if (accountName2 != null)
+                accounts.add(accountName2);
+        }
+        return accounts;
+    }
+
+    /**
+     * Removes the added account
+     */
+    public void removeAccount(String username) {
+        final UiObject removeAccount = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                "com.microsoft.identity.testuserapp:id/buttonRemoveAccount"
+        );
+        UiAutomatorUtils.obtainChildInScrollable("someone@contoso.com");
+        UiAutomatorUtils.handleInput("com.microsoft.identity.testuserapp:id/editTextUsername", username);
+        try {
+            removeAccount.click();
+        } catch (UiObjectNotFoundException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    /**
+     * Decode the sso token and look for the requested nonce
+     */
+    public void confirmNoncePresent() {
+        // fill the nonce
+        UiAutomatorUtils.obtainChildInScrollable("nonce");
+        UiAutomatorUtils.handleInput("com.microsoft.identity.testuserapp:id/editTextNonce", "testNonce");
```

Comment:
Looks like we're hard coding `testNonce` here...what's the utility of that?

Reply:
testNonce is a random string entered in the edit box to get the ssoToken. Once we get it, we decode it to see if the random nonce (testNonce) is present in the result as well.

Reply:
How do you plan to use this method? It seems this method figures out a nonce by itself, fills it in, and then asserts for its existence. What's the benefit of such a method?

I think instead we need to have the following two methods that replace this method:

```java
/**
* Acquires an SSO token using the provided nonce.
*/
public String acquireSsoToken(String nonce);

/**
* Decodes the SSO Token that was acquired earlier.
*/
public Map<String, ?> decodeSsoToken(String expectedNonce);
```

Note: I don't think we necessarily need that second method to decode the SSO Token since we can just do that without UI using the JWT Parser and therefore I suggest that you only implement one method here to just acquire the SSO Token.

Reply:
This method will be called in the testcase https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1561652
I have added this new testcase to send the nonce and verify the nonce after decoding. Even if we pass the nonce from the testcase, we would be doing the same thing. But yes, sending it as a param is a better option.

I agree with your suggestion that we can break it into 2 functions, but if we just keep the first one and skip the decodeSSoToken call, we will have to duplicate the logic executed when decodeSSoToken button is pressed (MainActivity.java in testuserApp has the decode base64 code)

I will be doing this in my testCase
1. Assign random string to nonce
2. Call acquireSSOToken(nonce)
3. Call decodeSSoToken(expectedNonce /*same as nonce*/)

Reply:
I guess what I'm saying is that we don't need to click the decode button to perform the actual decode since we can just do it in code ourselves; and yes the logic will be duplicated however, the test will become faster due to one less UI interaction. With that said, doing it via the UI is also fine..so feel free to add both methods.

Reply:
added changes in 060dcf041f3b0ff63dd5271687503dda4e4efc11